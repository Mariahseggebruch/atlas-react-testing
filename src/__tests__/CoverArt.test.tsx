// CoverArt.test.tsx
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import CoverArt from '../components/CoverArt'

it('renders CoverArt with an image', () => {
	const { container } = render(<CoverArt imageUrl="https://example.com/album-cover.jpg" />);
	expect(container).toMatchSnapshot(); 
  });
  
  it('renders CoverArt with a placeholder if no image is provided', () => {
	const { container } = render(<CoverArt imageUrl="" />);
	expect(container).toMatchSnapshot(); 
  });
  