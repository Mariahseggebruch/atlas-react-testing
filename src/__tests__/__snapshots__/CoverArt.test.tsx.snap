// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders CoverArt with a placeholder if no image is provided 1`] = `
<div>
  <div
    class="flex cover-art mb-6 w-full justify-center"
  >
    <img
      alt="Current Album Cover"
      class="rounded-md md:w-400 md:h-400"
      src=""
    />
  </div>
</div>
`;

exports[`renders CoverArt with an image 1`] = `
<div>
  <div
    class="flex cover-art mb-6 w-full justify-center"
  >
    <img
      alt="Current Album Cover"
      class="rounded-md md:w-400 md:h-400"
      src="https://example.com/album-cover.jpg"
    />
  </div>
</div>
`;
